test/NuGet.Clients.FuncTests/NuGet.CommandLine.FuncTest/Commands/RestoreCommandTests.cs

```diff
@@ -1168,6 +1168,211 @@ public async Task Restore_PackageNamespace_Fails()
+            <namespace id=""Contoso.*"" />             
+            <namespace id=""Test.*"" />
+        </packageSource>
+    </packageNamespaces>
+</configuration>");
+
+                var ContosoReal = new SimpleTestPackageContext()
+                {
+                    Id = "Contoso.A",
+                    Version = "1.0.0"
+                };
+                ContosoReal.AddFile("lib/net461/contosoA.dll");
+
+                await SimpleTestPackageUtility.CreateFolderFeedV3Async(
+                    contosoRepositoryPath,
+                    PackageSaveMode.Defaultv3,
+                    ContosoReal);
+
+                var ExternalContoso = new SimpleTestPackageContext()
+                {
+                    Id = "Contoso.A",  // This package exist both repo.
+                    Version = "1.0.0"
+                };
+                ExternalContoso.AddFile("lib/net461/externalA.dll");
+
+                await SimpleTestPackageUtility.CreateFolderFeedV3Async(
+                    externalRepositoryPath,
+                    PackageSaveMode.Defaultv3,
+                    ExternalContoso);
+
+                var ExternalB = new SimpleTestPackageContext()
+                {
+                    Id = "External.B",
+                    Version = "2.0.0"
+                };
+                ExternalB.AddFile("lib/net461/externalB.dll");
+
+                await SimpleTestPackageUtility.CreateFolderFeedV3Async(
+                    externalRepositoryPath,
+                    PackageSaveMode.Defaultv3,
+                    ExternalB);
+
+                Util.CreateFile(Path.GetDirectoryName(projectA.ProjectPath), "packages.config",
+@"<packages>
+  <package id=""Contoso.A"" version=""1.0.0"" targetFramework=""net461"" />
+  <package id=""External.B"" version=""2.0.0"" targetFramework=""net461"" />
+</packages>");
+
+                solution.Projects.Add(projectA);
+                solution.Create(pathContext.SolutionRoot);
+
+                // Act
+                var result = RunRestore(pathContext, _successExitCode, new string[] { "-source", $"{externalRepositoryPath};{contosoRepositoryPath}" });
+
+                // Assert
+                var contosoRestorePath = Path.Combine(projectAPackages, ContosoReal.ToString(), ContosoReal.ToString() + ".nupkg");
+                using (var nupkgReader = new PackageArchiveReader(contosoRestorePath))
+                {
+                    var allFiles = nupkgReader.GetFiles().ToList();
+                    // Assert correct Contoso package from Contoso repository was restored.
+                    Assert.Contains("lib/net461/contosoA.dll", allFiles);
+                }
+                var externalRestorePath = Path.Combine(projectAPackages, ExternalB.ToString(), ExternalB.ToString() + ".nupkg");
+                Assert.True(File.Exists(externalRestorePath));
+            }
+        }
+
+        [Fact]
+        public async Task Restore_PackageNamespace_PassNotEnoughSourceOption_Fails()
```

Comment:
Similar to the above, might be helpful to the style.

Reply:
Removed new tests from this file, please check above comment.